tests/test_forwarder.py

```diff
@@ -292,12 +338,247 @@ def _test_server(self, server):
+            remote_bind_address=(self.eaddr, self.eport),
+        )
+        self.assertEqual(server._transport.getpeername()[1], 22)
+
+    def test_unknown_argument_raises_exception(self):
+        """Test that an exception is raised when setting an invalid argument"""
+        with self.assertRaises(ValueError):
+            SSHTunnelForwarder(
+                self.saddr,
+                ssh_username=SSH_USERNAME,
+                ssh_password=SSH_PASSWORD,
+                remote_bind_address=(self.eaddr, self.eport),
+                i_do_not_exist=0
+            )
+
+    def test_more_local_than_remote_bind_sizes_raise_exception(self):
+        """
+        Test that when the number of local_bind_addresses exceed number of
+        remote_bind_addresses, an exception is raised
+        """
+        with self.assertRaises(ValueError):
+            SSHTunnelForwarder(
+                self.saddr,
+                ssh_username=SSH_USERNAME,
+                ssh_password=SSH_PASSWORD,
+                remote_bind_address=(self.eaddr, self.eport),
+                local_bind_addresses=[('127.0.0.1', self.eport),
+                                      ('127.0.0.1', self.randomize_eport())]
+             )
+
+    def test_localbindaddress_and_localbindaddresses_mutually_exclusive(self):
+        """
+        Test that arguments local_bind_address and local_bind_addresses cannot
+        be used together
+        """
+        with self.assertRaises(ValueError):
+            SSHTunnelForwarder(
+                (self.saddr, self.sport),
+                ssh_username=SSH_USERNAME,
+                ssh_password=SSH_PASSWORD,
+                remote_bind_address=(self.eaddr, self.eport),
+                local_bind_address=('127.0.0.1', self.eport),
+                local_bind_addresses=[('127.0.0.1', self.eport),
+                                      ('127.0.0.1', self.randomize_eport())]
+            )
+
+    def test_remotebindaddress_and_remotebindaddresses_are_exclusive(self):
+        """
+        Test that arguments remote_bind_address and remote_bind_addresses
+        cannot be used together
+        """
+        with self.assertRaises(ValueError):
+            SSHTunnelForwarder(
+                (self.saddr, self.sport),
+                ssh_username=SSH_USERNAME,
+                ssh_password=SSH_PASSWORD,
+                remote_bind_address=(self.eaddr, self.eport),
+                remote_bind_addresses=[(self.eaddr, self.eport),
+                                       (self.eaddr, self.randomize_eport())]
+            )
+
+    def test_no_remote_bind_address_raises_exception(self):
+        """
+        When no remote_bind_address or remote_bind_addresses are specified, a
+        ValueError exception should be raised
+        """
+        with self.assertRaises(ValueError):
+            SSHTunnelForwarder(
+                (self.saddr, self.sport),
+                ssh_username=SSH_USERNAME,
+                use_ssh_config=False
```

Comment:
In this case `use_ssh_config=False` can confuse the reader. This option is not uses for this test.